washboard-ui/packages/washboard-ui/tests/common/wasmcloud-instance.ts

```diff
@@ -0,0 +1,282 @@
+}
+
+/** Options for a given wasmCloud instance */
+export class WasmCloudInstanceOpts {
+  /** Override path to `wash` binary (by default this is `wash`) */
+  washBinPath?: string;
+  /** Whether to start Wash UI */
+  startWashUI?: boolean;
+  /** Output format to use for wash logs */
+  outputFormat?: WashOutputFormat;
+  /** Options for NATS usage with the wasmCloud instance */
+  nats?: WasmCloudInstanceNatsOpts;
+
+  /**
+   * An nkeys key (256-bit Ed25519 key) that the host uses to sign all invocations
+   * see: https://www.npmjs.com/package/nkeys.js
+   */
+  clusterSeed?: Uint8Array;
+
+  /** Labels (key, value) to apply to the host */
+  labels?: Array<[string, string]>;
+
+  /** Disable downloading of the wasmCloud host binary if not already installed */
+  disableHostDownload?: boolean;
+
+  /**
+   * Allow starting multiple additonal hosts on the same machine at the same time
+   * This is enabled by default.
+   * */
+  multiLocal?: boolean;
+  /**
+   * Whether to disable WADM
+   * see: https://github.com/wasmCloud/wadm
+   */
+  disableWadm?: boolean;
+
+  /** Options for `wash ui` invocation */
+  washUI?: ProcessExtra;
+
+  /** Whether to enable debug output/settings */
+  debug?: boolean;
+
+  /** Build a default instance of `WasmCloudInstanceOpts` */
+  static default() {
+    return {
+      // By default, most tests should use the *latest* version of the `wash ui` code in-tree,
+      // which means that starting `wash-ui` is unnecessary (use playwright's devServer instead)
+      startWashUI: false,
+      multiLocal: true,
+      debug: !!process.env.DEBUG ?? false,
+    };
+  }
+}
+
+/** Options to use while runnign `wash ui` */
+export interface ProcessExtra {
+  /**
+   * Port on which to start wash UI
+   * If this is not specified, one is randomly picked
+   * */
+  port?: number;
+}
+
+/** Process along with it's metadata */
+interface ProcessAndMeta {
+  stopped: boolean;
+  process?: ExecaChildProcess;
+  abort?: AbortController;
+  extra?: ProcessExtra;
+}
+
+/** Types of internal wash processes */
+enum WashProcessType {
+  Main = 'wash-main',
+  UI = 'wash-ui',
+}
+
+/**
+ * Instance of wasmCloud used for testing, generally managed with the WasmCloud Shell (`wash`)
+ *
+ * This instance is capable of spawning and controlling a wasmCloud host instance with `wash`, and
+ * exposes utility functions for performing common operations.
+ */
+export class WasmCloudInstance {
+  /** Options with which the wasmCloud instance should be created */
+  private opts: WasmCloudInstanceOpts;
+
+  /** Unique ID of this wash instance */
+  private _uuid: string;
+
+  /** Internal processes of this instance */
+  private processes: Map<WashProcessType, ProcessAndMeta> = new Map();
+
+  constructor(opts?: WasmCloudInstanceOpts) {
+    this.opts = opts ?? WasmCloudInstanceOpts.default();
+    this._uuid = uuidv1();
+  }
+
+  /** Return the UUID for this instance */
+  uuid(): string {
+    return this._uuid.slice();
+  }
```

Comment:
What's the need to do this?

Reply:
Ah thanks for pointing this out, I believe this was from when I was using the `Uint8Array` that comes out of the `uuid` package natively -- it's a string so I don't need this.

Will fix this and rebase!